### PR TITLE
fix: display real user profile

### PR DESCRIPTION
**Pull request by nukkk97**
Changing password and bio are not yet implemented.

### CODE REVIEW

File: src/app/profile/_components/ProfileHeader.tsx [hunk 1]
```diff
@@ -4,56 +4,60 @@ import { Avatar, Button, Dialog, DialogContent, DialogTitle, IconButton, Typogra
 import PaidIcon from '@mui/icons-material/Paid';
 import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
 import CloseIcon from '@mui/icons-material/Close';
+import { signOut, useSession } from "next-auth/react";
 
 type Props = {
-  username: string;
-  coinsLeft: number;
+  username: string | undefined;
```
Comment (yenting-biao) [inline src/app/profile/_components/ProfileHeader.tsx:10]: why undefined?

Comment (yenting-biao) [reply src/app/profile/_components/ProfileHeader.tsx:10]: If the user is not logged in, redirect may be a better method.

**Comment by nukkk97** [comment]
Now if you go to /profile without login, you will be redirected to /login.

File: src/app/profile/_components/ProfileHeader.tsx (hunk 1 above)
Comment (nukkk97) [reply src/app/profile/_components/ProfileHeader.tsx:10]: If you get to /profile without login, username will be undefined.
You can also protect the pages with next-auth/middleware, but for now I tend to keep it simple.

**Review by yenting-biao — APPROVED** [review]

**Pull request merged**